cmd/system-probe/modules/network_tracer.go

```diff
@@ -78,13 +80,78 @@ type networkTracer struct {
 	tracer       *tracer.Tracer
 	done         chan struct{}
 	restartTimer *time.Timer
+
+	// Intended to ensure (in compile time) we're implementing the SystemProbeServer interface
+	connectionserver.UnsafeSystemProbeServer
```

Comment:
The canonical way of doing this would be 
```go
var _ connectionserver.UnsafeSystemProbeServer = networkTracer{}
```

Reply:
Actually I have tried to use `UnimplementedSystemProbeServer` and `UnsafeSystemProbeServer` but both of them did not worked for me:
Cannot use '&networkTracer{}' (type *networkTracer) as the type connectionserver.UnsafeSystemProbeServer Type does not implement 'connectionserver.UnsafeSystemProbeServer' as some methods are missing: mustEmbedUnimplementedSystemProbeServer()

Am i missing something?

Reply:
I don't think you are missing anything. As the name suggest you "must embed UnimplementedSystemProbeServer". Like it's done here in some CWS code https://github.com/DataDog/datadog-agent/blob/8809daff3618cde141a010cf50df94ef31af21e5/pkg/security/module/server.go#L59

Reply:
Yes, I misinterpreted this. Embedding `UnimplementedSystemProbeServer` is just a different way to do what I suggested in my first comment. So you don't need to do that, but I would recommend you use `UnimplementedSystemProbeServer` instead of `UnsafeSystemProbeServer`.

Reply:
Hmm, are you certain that this change is an improvement? I noticed that when I employed the `UnimplementedSystemProbeServer`, making a mistake in altering the function name to something incorrect still allowed the code to compile. It successfully generated a function with an unimplemented error. On the other hand, when I utilized the `UnsafeSystemProbeServer` approach and included a non-existent function, it wouldn't compile at all. It flagged an error indicating that the function doesn't exist. Considering this, do you think it might be more advantageous for our specifics?

Reply:
Its an interface vs a struct. `UnsafeSystemProbeServer` is an interface, so you'd have to implement the one function it has.

I would embed `UnimplementedSystemProbeServer` for runtime errors, and use `var _ connectionserver.SystemProbeServer = networkTracer{}` for getting compile time errors.